### DISCUSSION TITLE
HM700 mit Parallel geschalteten Eingängen - Leistungschwankungen

### DISCUSSION THREAD

**Discussion by peff74** [hoylabs/OpenDTU-OnBattery#983] [Q&A] [answered]
Moin,
ich hatte in den letzten Tagen meinen HM700 wieder auf beide WR angeschlossen, dickes Y Kabel.
Lief jetzt sehr lange nur über einen der WR des HM700
Dabei fiel mir auf, dass es dadurch zu ziemlichen Schwingungen in manchen Leistungsbereichen kam.

Bisher hatte ich folgende Einstellungen im DPL
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/4b070fde-399b-4ab3-afa6-c84e1b1928d3)
Mit den parallel geschalteten musste ich auf eine Hysterese von 30 um Ruhe zu bekommen.

Bei genauerer Betrachtung fiel auf, dass ca. 1x pro Minute der WR mit den nun parallel geschalteten WR einen kurzen Leistungseinbruch von 50W hat. Sieht man quasi nur am Stromzähler selbst, da der pro Sekunde 2 Werte ausspuckt.

Wenn jetzt einer dieser Leitungseinbrüche vom IR über MQTT an die OpenDTU gelangt, wird nachgeregelt, und danach wieder zurück.
Folge --> Schwingen.
Besonders stark ist mir das bei Leistungen um 100-200W aufgefallen.

Habt Ihr sowas auch beobachtet?

**Comment by spcqike** [comment]
Du hast zwei WR parallel an einen WR angeschlossen? Du meinst Module/paneele, oder? 

Nein sowas stelle ich nicht fest. Ich kann mir auch nicht erklären, warum mehr Wp am Eingang zu mehr Schwankungen auf der AC Seite führen. 

Außer …… sind deine Module unterschiedlich verschattet? Vielleicht liegen sie ungünstig sodass der MPPT stärker schwingt um den MPP zu ermitteln. Meine Module sind alle identisch ausgerichtet

**Reply by peff74** [reply]
Vielleicht hätte ich es genauer beschreiben sollen.
Der HM700 hängt an einem Akku, der über einen Victron LR gespeist wird. --> OpenDTU-onBatt Prinzip ;-)
Bisher hatte ich lange nur einen Anschluss des HM700, also bis 350W genutzt.
Dies hatte ich nun umgebaut, mit einem Y Kabel um beide Eingänge/WR des HM700 zu nutzen.
Beide Eingänge zeigen sehr ähnliche Spannungs- und Stromstärken, somit ist die Verkabelung O.K.

Kann es vielleicht mit dem internen Tracking der WR im HM700 zu tun haben?
Bei den Victrons gibt es ja auch alle 10min solche Leistungseinbrüche.

**Reply by spcqike** [reply]
Achso. Ja das macht es etwas klarer. Leider könnten Gründe dafür dennoch sehr vielfältig sein.

Was für eine Batterie hast du? ggf. ist diese zu klein, sodass die DC Spannung stark und schnell schwankt? Das wiederum würde/könnte sich mit den MPPT im Wechselrichter haken. 

> Beide Eingänge zeigen sehr ähnliche Spannungs- und Stromstärken, somit ist die Verkabelung O.K.

Ist das so? Es gab bereits schlechte Verbindungen, bei denen man mit der vorhandenen Messtechnik (Wechselrichter und Anzeige per DTU) keine Auffälligkeiten gesehen hat. Überprüf doch ggf. einfach nochmal den Sitz der Verbindung oder tausche das Y-Kabel, wenn du ein zweites hast, einfach mal.

> Kann es vielleicht mit dem internen Tracking der WR im HM700 zu tun haben?

Klar, kann sein. die Hoymiles sind nicht für Batteriebetrieb gedacht, auch wenn es doch recht gut funktioniert. Ein Wechselrichter, der einen Batteriemodus hat, kann den MPPT ausschalten und hat dadurch weniger PRobleme.

**Comment by peff74** [comment]
ich habe mir das jetzt noch einmal ein wenig genauer angeschaut.
Es sieht so aus, als ob der/die WR im niedrigen Leistungsbereich nur diese Schwankungen haben.

Auf dem Bild sieht man bei einer benötigten Leistung von ~100W viele kleine Schwankungen.
Sobald die benötigte Leistung etwas steigt ~200W, so ab 06:12 wird es schon deutlich ruhiger.

- „Energie aus Netz“ ist das, was der Stromzähler sieht, im Sekundentakt.
- „Energie aus Solar“ zeigt die gelieferte Leistung des/der WR. Ist hier schon deutlich glatter, da ich OpenDTU mittlerweile einen Durchschnitt der letzten 3 Sekunden des Stromzählers als Stromzählerwert liefere.

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/81a16423-e874-46c8-851b-9a56728a210c)

Ab ~300 läuft es dann sehr glatt so wie ich es kenne.
Wahrscheinlich ist mir das früher nie aufgefallen, da wenn ich nur mit einem Anschluss gefahren bin und somit der eine WR nur sehr selten unter 100W kam.

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/39e0b1f5-c33a-4fa2-888a-0b50abe71cdb)


Wenn das ein „normales“ Verhalten ist, sei’s drum.
Dann bügele ich die Zählerwerte, bei einer WR Leistung unter 200W einfach ein wenig glatt, bevor sie zur OpenDTU-onBatt kommen und gut is.

**Reply by spcqike** [reply]
mir fällt es schwer, dein Problem nachzuvollziehen. Wo genau siehst du diese Schwankungen? Welche Verbraucher hast du aktiv? Welche Einstellungen allgemein? Bist du sicher, dass das "Gezappel" aus dem Einspeisung her rührt und nicht von einem dynamischen Verbraucher?

Ein Computer bspw. kann im ms Takt die Leistungsaufnahme stark variieren. Da regelt sich ne DTU natürlich kaputt dran, wenn sie versuchen soll, das zeitnah auszugleichen.
mein Server bspw. schwankt im Idle bereits um gut 30W. 
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/95851bfc-bd9f-4544-83bc-cabf0911ae37)

bei mehreren sich überlagernden Verbrauchern, kann sich das natürlich stärker aufschaukeln.

Wie sieht es denn aus, wenn du den DPL deaktivierst und ein einfaches Limit setzt? also konstant einspeist? bzw. den Inverter einfach mal stoppst. ist dein Netzbezug denn dann "glatt"?

**Reply by peff74** [reply]
Ich hatte da ganze schon einmal mit fest eingestellten 120W getestet und alle Sicherungskreise des Hauses nach und nach herausgenommen. Da ich mir keinen Reim auf diese Schwankung machen konnte.
Zum Schluss war nur noch der WR selbst mit den ESP Komponenten und das Gartenhaus am Netz.
EDV läuft über einen anderen Kreis/Zähler.

Das Ergebnis war ein Stromzähler mit -120W, der ab und an ~-75W anzeigte. Alle paar Sekunden.
Somit kommt's vom System.
Also entweder "normal" bei niedriger Leitung oder das Ding hat 'ne Macke.
Vermutung liegt bei ersterem.
Wollte nur ggf. auf Erfahrungen hier zurückgreifen.

**Comment by Manos1966** [comment]
Sind deine Kabel zum HM-700 gleich lang?
Hast du die gleiche Amperes Leistung auf beide Seiten?
Siehe Beispiel wo es NICHT der Fall ist:
![901 A difference](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/7edfc3c5-6951-481b-8c0d-7a612a02d23d)

**Reply by peff74** [reply]
würde sagen sieht bei mir sauber aus:
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/7360c73a-ae65-487b-8203-c78b2c31782a)

**Comment by peff74** [comment]
Heute Morgen ist zum ersten Mal seit langem der Solar Passthrough aktiv.
Auch hier sieht man einen deutlichen Unterschied zu parallel geschalteten Eingängen gegenüber nur einem angeschlossenen Eingang.
Mit nur einem Eingang:
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/2ca7d0ed-a3b3-4199-90ae-f8796ed51163)
Mit parallel geschalteten Eingängen 
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/f7ef1f8e-06b7-45dc-84a0-5cf8b813954e)

Mich würde grundsätzlich interessieren, ist das normal so, oder liegt da ein Defekt vor?

**Reply by peff74** [reply]
hier noch schön zu sehen, ab ~250W kehrt Ruhe ein.

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/f47e30a4-89d5-412c-8a50-598f9463549e)

**Comment by peff74** [comment, marked as answer]
Ich habe das Thema noch weiter eingekreist.
Das Problem besteht nur im Bereich 15-35% relatives Limit.
Wobei es um 25% am stärksten auftritt.
Man kann es deutlich in der Live-Ansicht der DTU sehen.
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/f22424ee-48b7-470d-a5c6-5e10b1dacd0b)
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/6af2fa49-9024-4fa0-a312-3fde06bc55c5)
Gleiches Limit und Leistungs-Sprünge von fast 20W pro Kanal von einer Sekunde zur anderen.

 Das diese das System durcheinander bringt, ist klar.

**Reply by Manos1966** [reply]
Danke fuer die ganze Arbeit @peff74 
ich werde deine Analyse in das Wiki addieren, ein weiterer (externer) Grund warum die OpenDTU-OnBattery so viel Mühe hat die Leistung zu regeln 👍 

[Dynamic Power Limiter Limitations](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Dynamic-Power-Limiter-Limitations#fluctuations-in-inverter-power)

**Reply by peff74** [reply]
Wichtig zu erwähnen ist, dass Problem tritt nur auf bei parallel geschalteten Eingängen!
Und ich kann es bisher nur für den HM700 bestätigen, und für mein Device. (vielleicht hat das Ding ja auch ne Macke?)
Habe noch einen HM600 liegen und werde es mit ihm die Tage auch mal testen.

Mich wundert, dass das bisher noch keinem Aufgefallen is....

**Reply by spcqike** [reply]
> Mich wundert, dass das bisher noch keinem Aufgefallen is....

Wie du sagst, vielleicht betrifft es gar nicht alle Modelle oder gar alle Firmware-Stände einer Modellreihe. Vielleicht hängen auch andere Faktoren mit drin wie die Batterie/-belastung und der Spannungseinbruch. Vielleicht regelt der MPPT genau in dem Spannungsbereich verstärkt?

Wenn es an der Leistung liegt, wie viele Leute fahren genauso wie du 175W, wo das Problem auftaucht, für wie lange, dass es einem auffallen könnte? :) 

Es gibt wahrscheinlich einfach viel zu viele Parameter und Möglichkeiten. Auch wenn es systematisch ist (Software Hoymiles), kann es doch nur in Einzelfällen auftreten, wie bei dir.

Ist da Verhalten mit nur einem Eingang analog? Also bei gleicher prozentualer Belastung (25%, also 87.5W für den einzelnen Eingang) bzw. bei gleicher prozentualer Belastung (50% für einen Eingang, also wieder 175W gesamt)

**Reply by peff74** [reply]
Das Problem tritt ja nicht nur bei 175W auf, sondern im ganzen Spektrum um 15-35% und das sind ja nun 105-245W und das lieg ja im Bereich vieler Grundverbräuche.
Habe nun mit dem HM600 getestet.
Schwankt auch im Bereich 15-35%, aber bei weitem nicht so viel ~6-7W pro Kanal.
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/67f251d8-5ecb-4d01-b0e9-755e2470cbb4)
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/8e677a29-e29d-4133-b4f9-375233a5a13f)

**Reply by Manos1966** [reply]
@peff74 soll ich auch die SerNr beim anderen Bild entfernen?

**Reply by peff74** [reply]
hast Du die Rechte dazu?
Is glaube ich nicht so wild, oder?

**Reply by Manos1966** [reply]
schon geschehen in das Wiki... dein eigenes Post hier oben muss du aber selber editieren.
Wild ist es nicht, du wirst sowieso deine HMs nicht in die chinesische cloud stellen 😉 
Aber es ist gute "Disziplin" solche Daten zu entfernen.
